src/components/tabController/TabBarItem.tsx

```diff
@@ -226,6 +226,9 @@ export default function TabBarItem({
         style={_style}
         onLayout={onLayout}
         testID={testID}
+        accessible
+        accessibilityRole="tab"
+        accessibilityState={{selected: selectedIndex === index}}
```

Comment:
I think you can use `currentPage.value` to remove most of the code in the PR
```suggestion
        accessibilityState={{selected: currentPage.value === index}}
```

Reply:
Maybe it's the re-selection, I'll re-test

Reply:
Changed. I added `useAnimatedReaction` in the `TabBarItem` that will change its state only when become selected or unselected.